src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -386,22 +393,12 @@ public IEnumerable<X509Extension> Extensions
             }
         }
 
-        public string GetNameInfo(X509NameType nameType, bool forIssuer)
-        {
-            CertNameType certNameType = MapNameType(nameType);
-            CertNameFlags certNameFlags = forIssuer ? CertNameFlags.CERT_NAME_ISSUER_FLAG : CertNameFlags.None;
-            CertNameStrTypeAndFlags strType = CertNameStrTypeAndFlags.CERT_X500_NAME_STR | CertNameStrTypeAndFlags.CERT_NAME_STR_REVERSE_FLAG;
-
-            int cchCount = Interop.crypt32.CertGetNameString(_certContext, certNameType, certNameFlags, ref strType, null, 0);
-            if (cchCount == 0)
-                throw Marshal.GetLastWin32Error().ToCryptographicException();
-
-            StringBuilder sb = new StringBuilder(cchCount);
-            if (Interop.crypt32.CertGetNameString(_certContext, certNameType, certNameFlags, ref strType, sb, cchCount) == 0)
-                throw Marshal.GetLastWin32Error().ToCryptographicException();
-
-            return sb.ToString();
-        }
+        public unsafe string GetNameInfo(X509NameType nameType, bool forIssuer) =>
+            Interop.crypt32.CertGetNameString(
+                _certContext,
+                MapNameType(nameType),
+                forIssuer ? CertNameFlags.CERT_NAME_ISSUER_FLAG : CertNameFlags.None,
+                (int)(CertNameStrTypeAndFlags.CERT_X500_NAME_STR | CertNameStrTypeAndFlags.CERT_NAME_STR_REVERSE_FLAG));
```

Comment:
If you decide to do another iteration, I'd be fine with this switching to CertNameStringType to let the parameter go back to not requiring an explicit cast to call it.

Not sure why this one was different than the other two.